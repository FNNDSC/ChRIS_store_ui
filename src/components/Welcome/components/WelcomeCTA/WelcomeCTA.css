.welcome-cta {
  font-size: 1.4em;
}

.welcome-cta-img {
  /* color */
  background: #393f44;
  background: var(--pf-black-800);
  background: linear-gradient(135deg #4d5258, #393f44);
  background: linear-gradient(135deg, var(--pf-black-700), var(--pf-black-800));
  background: url('../../../../assets/img/banner_chris-users.png') #393f44;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  color: #fafafa;
  color: var(--pf-black-100);

  /* display */
  padding: 3em 30px;

  /* font */
  text-shadow: 3px 3px 6px #030303;
  text-shadow: 3px 3px 6px var(--pf-black);

  /* vertically align text in div */
  display: flex;
  justify-content: center;
  flex-direction: column;
}

.welcome-cta-header {
  /* font */
  font-size: 1.3em;
  line-height: 1.3;
}

.welcome-cta-featured {
  /* color */
  background: #0088ce;
  background: var(--pf-blue-400);
  background: linear-gradient(#39a5dc, #0088ce);
  background: linear-gradient(var(--pf-blue-300), var(--pf-blue-400));

  /* display */
  min-height: 300px;
  padding-bottom: 1.5em;
}

.welcome-cta-featured-desc {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);

  /* display */
  padding: 1.5em 1em;

  /* font */
  text-align: center;
  font-weight: 300;
}

.welcome-cta .welcome-cta-featured-container {
  /* display */
  display: flex;
  width: 300px;
  width: fit-content;

  /* flex */
  flex-direction: column;
}

.welcome-cta-featured-container > * {
  flex: 1;
}

.welcome-scroll-caret {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);

  /* display */
  display: none;

  /* font */
  font-size: 2em;
  text-align: center;
}

.welcome-user-cta {
  /* display */
  padding: 6em 2em;
  padding-bottom: 8em;
}

.welcome-user-cta-header {
  /* display */
  margin-bottom: 0.5em;

  /* font */
  font-size: 2em;
}

/* ============================== */
/* ---------- ANIMATIONS -------- */
/* ============================== */

@keyframes fadein {
  0% { opacity: 0; }
  60% { opacity: 0; }
  100% { opacity: 1; }
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 480px)  {
  .welcome-cta-img {
    /* display */
    padding: 3em 2em;
  }

  .welcome-cta-header {
    /* font */
    font-size: 1.9em;
  }
}

@media (min-width: 768px)  {
  .welcome-cta-img {
    /* display */
    min-height: 300px;
    /* fill up the entire page except for the features (300px) and navbar (66px) */
    height: calc(100vh - 325px - 66px);
    padding: 0;
  }

  .welcome-cta-header {
    /* font */
    font-size: 2.5em;
  }

  .welcome-cta-featured {
    /* display */
    min-height: 325px;
    padding-bottom: 0;
  }

  .welcome-cta .welcome-cta-featured-container {
    /* display */
    width: auto;

    /* flex */
    flex-direction: row;
  }

  .welcome-cta-header { padding: 0 8%; }

  .welcome-scroll-caret {
    /* display */
    display: block;

    /* other */
    animation: fadein 2s;
  }

  .welcome-user-cta {
    width: 70%;
  }
}
