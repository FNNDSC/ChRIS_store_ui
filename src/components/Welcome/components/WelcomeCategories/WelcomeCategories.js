import React, { Component } from 'react';
import sampleCategories from './sampleCategories';
import './welcomeCategories.css';

import WelcomeCategory from '../WelcomeCategory/WelcomeCategory';

class WelcomeCategories extends Component {
  constructor(props) {
    super(props);

    this.state = {
      categories: sampleCategories,
    };
  }

  render() {
    return (
      <div className="welcome-categories">
        <div className="welcome-categories-header">Browse our catalog of tools ready to deploy to ChRIS</div>
        <div className="welcome-categories-container">
          {
            this.state.categories.map(category => (
              <WelcomeCategory category={category} key={category.name} />
            ))
          }
        </div>
      </div>
    );
  }
}

export default WelcomeCategories;
