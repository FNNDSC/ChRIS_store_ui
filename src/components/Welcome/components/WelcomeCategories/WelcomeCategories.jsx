import React from 'react';
import sampleCategories from './sampleCategories';
import styles from './welcomeCategories.module.css';
import Category from '../WelcomeCategory/WelcomeCategory';

const WelcomeCategories = () => {
  const categories = sampleCategories ;

  return (
    <div className={styles.welcomeCategories}>
      <div className={styles.welcomeCategoriesHeader}>
        <span className="tag">TOOLS</span><br/>
        <h1>Browse our Catalog of <br/> Tools ready to Deploy to ChRIS</h1>
        <p>
          ChRIS is an democratizing the development and usage of image processing
          <br/> software with an ecosystem following common standards.
        </p>
      </div>
      <div className={styles.welcomeCategoriesContainer }>
        {categories.map((category) => (
          <Category category={category} key={category.name} />
        ))}
      </div>
    </div>
  );
};

export default WelcomeCategories;
