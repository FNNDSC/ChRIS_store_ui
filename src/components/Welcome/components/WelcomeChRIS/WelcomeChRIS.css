.welcome-chris {
  /* color */
  background: #39a5dc;
  background: var(--pf-blue-300);

  /* display */
  min-height: 400px;
  padding: 2em;

  /* position */
  position: relative;
}

.welcome-chris .row > div {
  /* display */
  padding: 1em;

  /* flex */
  flex: 1;
}

.welcome-chris-text {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);
}

.welcome-chris-text-container > div {
  font-weight: 400;
  font-size: 3em;
}

.welcome-chris-text-container > p { font-size: 1.3em; }

.welcome-chris-video-column {
  /* display */
  display: flex;

  /* flex */
  flex-direction: column;
  justify-content: center;
}

.welcome-chris-video {
  /* display */
  width: 100%;
  height: 200px;
}

.welcome-chris-video-desc {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);

  /* display */
  padding: 1em 1em 0 0;
}

.welcome-chris-btn {
  /* display */
  display: block !important;
  width: 100%;
}

.welcome-chris-banner { display: none; }

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 420px) {
  .welcome-chris-btn-row > div { display: inline-block; }

  .welcome-chris-btn { width: auto; }
}

@media (min-width: 480px)  {
  .welcome-chris-video { height: 50vw; }
}

@media (min-width: 768px)  {
  .welcome-chris-video {
    /* display */
    width: 45vw;
    height: 25.3vw;
  }

  .welcome-chris-more-btn {
    /* display */
    width: 208.5px;

    /* other */
    float: right;
  }

  .welcome-chris > .row:first-child { flex-direction: row-reverse; }
}

@media (min-width: 1108px) {
  .welcome-chris-video {
    /* display */
    width: 500px;
    height: 281.25px;
  }

  .welcome-chris-text-container {
    /* display */
    padding: 0 2em 1.1em 2em;
    width: 33vw;
  }

  .welcome-chris-banner {
    /* display */
    display: block;

    /* position */
    position: absolute;
    top: 0;
    right: 5em;
  }
}

@media (min-width: 1267px) {
  .welcome-chris-text-container { width: auto; }
}
