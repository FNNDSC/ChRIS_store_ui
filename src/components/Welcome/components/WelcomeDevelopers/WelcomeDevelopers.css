.welcome-developers {
  /* display */
  padding: 6em 2em;
  padding-bottom: 8em;

  /* font */
  font-size: 1.4em;
}

.welcome-developers-header {
  /* font */
  font-size: 2em;
  line-height: 1.3;
}
