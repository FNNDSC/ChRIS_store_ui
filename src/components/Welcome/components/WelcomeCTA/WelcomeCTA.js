import React from 'react';
import { Icon } from 'patternfly-react';
import './WelcomeCTA.css';
import WelcomeFeature from '../WelcomeFeature/WelcomeFeature';

/* plugin images */
import freesurferLogo from '../../../../assets/img/plugins/freesurfer_90.png';
import tensorflowLogo from '../../../../assets/img/plugins/tensorflow_90.png';
import fslLogo from '../../../../assets/img/plugins/fsl_90.png';
import rhoanaLogo from '../../../../assets/img/plugins/rhoana_90.png';
import slicerLogo from '../../../../assets/img/plugins/slicer_90.png';

const WelcomeCTA = () => (
  <div className="welcome-cta">
    <div className="welcome-cta-img">
      <div className="row welcome-cta-header">
        Accessible medical imaging using the latest
        research innovations, backed by cloud-based
        computing power.
      </div>
    </div>
    <div className="welcome-cta-featured">
      <div className="welcome-cta-featured-desc">
        Apps available in the ChRIS store include:
      </div>
      <div className="row welcome-cta-featured-container">
        <WelcomeFeature name="FreeSurfer" img={freesurferLogo} />
        <WelcomeFeature name="TensorFlow" img={tensorflowLogo} />
        <WelcomeFeature name="FMRIB Software Library" img={fslLogo} />
        <WelcomeFeature name="RhoANA" img={rhoanaLogo} />
        <WelcomeFeature name="Slicer" img={slicerLogo} />
      </div>
      <div className="welcome-scroll-caret">
        <Icon name="angle-down" />
      </div>
    </div>
    <div className="row no-flex">
      <div className="welcome-user-cta">
        <div className="welcome-user-cta-header">
          Focus on your data. Not the tools.
        </div>
        <div className="text-light">
          You need to run analyses on data, view the results,
          create visualizations, collaborate on your findings.
          Not build an infrastructure and become a software developer.
        </div>
        <br />
        <div className="text-light">
          The ChRIS platform provides a common infrastructure to
          which you can deploy...
        </div>
      </div>
    </div>
  </div>
);

export default WelcomeCTA;
