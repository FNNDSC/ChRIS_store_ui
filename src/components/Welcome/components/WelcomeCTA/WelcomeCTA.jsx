import React from 'react';
import './WelcomeCTA.css';
import WelcomeFeature from '../WelcomeFeature/WelcomeFeature';

/* plugin images */
import infantLogo from '../../../../assets/img/plugins/infant_90.png';
import freesurferLogo from '../../../../assets/img/plugins/freesurfer_90.png';
import biomediaLogo from '../../../../assets/img/plugins/biomedia_90.png';
import antLogo from '../../../../assets/img/plugins/ant_90.png';
import civetLogo from '../../../../assets/img/plugins/civet_90.png';

const WelcomeCTA = () => {
  // Object to make fetching from build/external easier later
  const features = [
    { name: 'Fastsurfer', url: 'https://chrisstore.co/plugin/44', img: freesurferLogo },
    { name: 'Civet', url: 'https://chrisstore.co/plugin/2', img: civetLogo },
    { name: 'Infant FreeSurfer', url: 'https://chrisstore.co/plugin/78', img: infantLogo },
    { name: 'IRTK Reconstruction', url: 'https://chrisstore.co/plugin/85', img: biomediaLogo },
    { name: 'N4 Bias Field Correction', url: 'https://chrisstore.co/plugin/77', img: antLogo },
  ];

  return (
    <>
      <div className="welcome-cta">
        <article>
          <div id="welcome-cta-header">
            <h1>ChRIS Store</h1>
            <p>
              Accessible medical imaging using the latest research innovations, backed by
              cloud-based computing power.
            </p>
          </div>
        </article>

        <article>
          <div className="welcome-cta-featured">
            {/* <div style={{ opacity: 0.75, color: 'white', textAlign: 'center' }}>
            <h2>Apps available in the ChRIS Store</h2>
          </div> */}
            <div className="row welcome-cta-featured-container">
              {features.map((feature) => (
                <WelcomeFeature key={feature.name} {...feature} />
              ))}
            </div>
          </div>
        </article>
      </div>
    </>
  );
};

export default WelcomeCTA;
