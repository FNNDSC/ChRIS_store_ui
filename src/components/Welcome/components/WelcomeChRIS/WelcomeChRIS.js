import React from 'react';
import { Button } from 'patternfly-react';
import './WelcomeChRIS.css';
import chrisBanner from '../../../../assets/img/chris_logo-ribbon.png';

const WelcomeChRIS = () => (
  <div className="welcome-chris">
    <div className="row">
      <div className="welcome-chris-text">
        <div className="welcome-chris-text-container">
          <div>
            What is ChRIS?
          </div>
          <p>
            ChRIS is an open source framework that utilizes cloud technologies to
            democratize medical analytics application development and enables
            healthcare organizations to keep owning their data while benefiting from
            public cloud processing capabilities.
          </p>
          <p>
            ChRIS uses Docker/containers, Kubernetes/OpenShift, and a set of other
            technologies to standardize healthcare application development.
          </p>
        </div>
      </div>
      <div className="welcome-chris-video-column">
        <div>
          <iframe
            title="ChRIS video"
            className="welcome-chris-video"
            src="https://www.youtube-nocookie.com/embed/dyFQD87jU68"
            frameBorder="0"
            allow="autoplay; encrypted-media"
            allowFullScreen
          />
          <p className="welcome-chris-video-desc">
            {'"'}Creating ChRIS{'"'} is an upcoming video series by Red Hat films that
            focuses on the ChRIS project - its purpose, development, milestones.
          </p>
        </div>
      </div>
    </div>
    <div className="row welcome-chris-btn-row">
      <div>
        <Button
          bsStyle="default"
          bsSize="large"
          className="welcome-chris-btn welcome-chris-more-btn"
        >
          Learn more
        </Button>
      </div>
      <div>
        <Button
          bsStyle="default"
          bsSize="large"
          className="welcome-chris-btn"
        >
          Don{"'"}t have access to ChRIS?
        </Button>
      </div>
    </div>
    <span>
      <img src={chrisBanner} alt="" className="welcome-chris-banner" />
    </span>
  </div>
);

export default WelcomeChRIS;
