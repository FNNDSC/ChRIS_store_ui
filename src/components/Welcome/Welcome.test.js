import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './Welcome';

/* components */
import WelcomeCTA from './components/WelcomeCTA/WelcomeCTA';
import WelcomeCategories from './components/WelcomeCategories/WelcomeCategories';
import WelcomeChRIS from './components/WelcomeChRIS/WelcomeChRIS';
import WelcomeDevelopers from './components/WelcomeDevelopers/WelcomeDevelopers';

describe('Welcome', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Welcome />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render WelcomeCTA component', () => {
    expect(wrapper.containsMatchingElement(<WelcomeCTA />)).toEqual(true);
  });

  it('should render WelcomeCategories component', () => {
    expect(wrapper.containsMatchingElement(<WelcomeCategories />)).toEqual(true);
  });

  it('should render WelcomeChRIS component', () => {
    expect(wrapper.containsMatchingElement(<WelcomeChRIS />)).toEqual(true);
  });

  it('should render WelcomeDevelopers component', () => {
    expect(wrapper.containsMatchingElement(<WelcomeDevelopers />)).toEqual(true);
  });
});
