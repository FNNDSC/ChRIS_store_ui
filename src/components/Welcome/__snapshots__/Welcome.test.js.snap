// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WelcomeCTA />,
        <WelcomeCategories />,
        <WelcomeChRIS />,
        <WelcomeDevelopers />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WelcomeCTA />,
          <WelcomeCategories />,
          <WelcomeChRIS />,
          <WelcomeDevelopers />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
